lms/events/__init__.py

```diff
@@ -0,0 +1,77 @@
+from dataclasses import asdict, dataclass, field
+from typing import List, Optional
+
+from pyramid.events import subscriber
+from pyramid.request import Request
+
+from lms.models import EventType, LtiLaunches
+from lms.services import EventService, LTIRoleService
+
+
+@dataclass
+class BaseEvent:  # pylint:disable=too-many-instance-attributes
+    request: Request
+    type: EventType.Type
+
+    user_id: Optional[int] = None
```

Comment:
I don't know whether the `Option` here has any kind of effect or not? In the case of `dataclass` setting this to `None` means it's optional.

I genuinely don't really know. I suppose it certainly doesn't hurt?

Reply:
I have some typing checking tool installed ([pyright](https://github.com/microsoft/pyright)) and  it doesn't like int = None

Everything seems to work and this is more pedantically right, the best version of right!